backend/utils/room_utils.py

```diff
@@ -0,0 +1,290 @@
+    ) -> dict:
+        """stores the room profile data for the sidebar
+
+        Args:
+            member_id (str): member_id of the current user
+            room (dict): room object data
+            org_members (list): list of all members in the organization
+
+        Returns:
+            dict: key value pair of room profile
+        """
+        room_profile = {}
+        if room["plugin_name"] == "dm":
+            room_members = await self.__get_room_members(member_id, room, org_members)
+        room_profile["room_id"] = room["_id"]
+        room_profile["room_url"] = f"/{room['plugin_name']}/{room['_id']}"
+        room_profile["room_name"] = (
+            await self.__get_dm_room_name(room_members)
+            if room["plugin_name"] == "dm"
+            else room["room_name"]
+        )
+        room_profile["image_url"] = (
+            await self.__get_dm_room_image_url(room_members)
+            if room["plugin_name"] == "dm"
+            else ""
+        )
+        return room_profile
+
+    async def __get_joined_rooms(
+        self, member_id: str, user_rooms: list, org_members: list
+    ) -> dict:
+        """gets the profiles for all rooms for the sidebar
+
+        Args:
+            member_id (str): member_id of the current user
+            user_rooms (list): list of all rooms of the current user
+            org_members (list): list of all members in the organization
+
+        Returns:
+            dict: contains data as key value store of room profiles
+            sample_data: {"rooms": list,
+                          "starred_rooms":list
+                          }
+        """
+        rooms = []
+        starred_rooms = []
+        for room in user_rooms:
+            room_profile = await self.__get_room_profile(member_id, room, org_members)
+            rooms.append(room_profile)
+            if room["room_members"].get(member_id, {}).get("starred", False) is True:
+                starred_rooms.append(room_profile)
+        return {"rooms": rooms, "starred_rooms": starred_rooms}
+
+    async def __get_public_rooms(
+        self, member_id: str, org_id: str, org_members: list
+    ) -> dict:
+        """gets the public rooms for the sidebar
+
+        Args:
+            member_id (str): member_id of the current user
+            org_members (list): list of all members in the organization
+
+        Returns:
+            dict: contains data as key value store of room profiles
+            sample_data: key value pair of room profile
+        """
+        rooms = []
+        public_rooms = await get_org_rooms(org_id, is_private=False)
+        if public_rooms:
+            for room in public_rooms:
+                room_profile = await self.__get_room_profile(
+                    member_id, room, org_members
+                )
+                rooms.append(room_profile)
+        return rooms
+
+    async def format(self, org_id: str, member_id: str, plugin: str) -> dict:
```

Comment:
This method is 😻.
I just suggest we could find a better name.

Reply:
modified to format_data